packages/core-components/src/components/Link/Link.tsx

```diff
@@ -46,12 +47,44 @@ const useStyles = makeStyles(
 
 export const isExternalUri = (uri: string) => /^([a-z+.-]+):/.test(uri);
 
+/**
+ * Get the app base path from the configured app baseUrl.
+ *
+ * The returned path does not have a trailing slash.
+ */
+const getBasePath = (configApi: Config) => {
+  let { pathname } = new URL(
+    configApi.getOptionalString('app.baseUrl') ?? '/',
+    'http://dummy.dev', // baseUrl can be specified as just a path
+  );
+  pathname = pathname.replace(/\/*$/, '');
+  return pathname;
+};
+
 export type LinkProps = MaterialLinkProps &
   RouterLinkProps & {
     component?: ElementType<any>;
     noTrack?: boolean;
   };
 
+export const useResolvedPath = (uri: LinkProps['to']) => {
+  let resolvedPath = String(uri);
+
+  try {
+    const configApi = useApi(configApiRef);
```

Comment:
Reading this I'm a bit uncertain of which function calls that actually throws, would it be possible to just keep the ones that we catch and the assignment inside this block?

Reply:
Extracted to a separated hook called `useConfigApi`, the try-catch was used to return an empty config when the Config API is not Implemented. As @iamEAP said, the only cases there would be no Config API are in tests and in storybook stories, and in those cases, it's unlikely that callers would rely on this subpath behavior.